ddtrace/contrib/aws_lambda/patch.py

```diff
@@ -32,21 +32,25 @@ def _crash_flush(_, __):
 
 def _handle_signal(sig, f):
     """
-    Wraps the given signal with a previously defined, if exists.
+    Wraps the given signal with a previously defined signal, if
+    it's callable and is different than the function `f` to handle.
+    Else, it doesn't wrap and just sets the function to handle as `f`.
     This to avoid our signals overriding existing ones.
 
-    Returns the handler of the wrapped signal.
+    Returns a signal of type `sig` with function `_f`
     """
-    old_signal = None
-    if callable(signal.getsignal(sig)):
-        old_signal = signal.getsignal(sig)
+    old_signal = signal.getsignal(sig)
+    if not callable(old_signal) or old_signal == f:
+        old_signal = None
```

Comment:
Can we simplify the logic into something like: 

```suggestion
    if not callable(old_signal) or old_signal == f:
        return signal.signal(sig, f)
```
This is the intended result if there is already a wrapped f on the signal, correct?

Reply:
Yes, I changed it to your suggestions since it's more readable, thanks for the feedback!